product_docs/docs/pgd/5/ddl/ddl-role-manipulation.mdx

```diff
@@ -0,0 +1,54 @@
+---
+title: DDL and role manipulation statements
+navTitle: Roles
+---
+
+Users are global objects in a PostgreSQL instance, which means they span
+multiple databases while PGD operates on an individual database level. This means
+that role manipulation statement handling needs extra thought.
+
+PGD requires that any roles that are referenced by any replicated DDL must
+exist on all nodes. The roles don't have to have the same grants,
+password, and so on, but they must exist.
+
+PGD replicates role manipulation statements if [`bdr.role_replication`](/pgd/latest/reference/pgd-settings#bdrrole_replication) is
+enabled (default) and role manipulation statements are run in a PGD-enabled
+database.
+
+The role manipulation statements include the following:
+
+-   `CREATE ROLE`
+-   `ALTER ROLE`
+-   `DROP ROLE`
+-   `GRANT ROLE`
+-   `CREATE USER`
+-   `ALTER USER`
+-   `DROP USER`
+-   `CREATE GROUP`
+-   `ALTER GROUP`
+-   `DROP GROUP`
+
+In general, either:
+
+-   Configure the system with `bdr.role_replication = off` and
+    deploy all role changes (user and group) by external orchestration
+    tools like Ansible, Puppet, and Chef or explicitly replicated by
+    `bdr.replicate_ddl_command()`.
+
+-   Configure the system so that exactly one PGD-enabled database
+    on the PostgreSQL instance has `bdr.role_replication = on` and run all
+    role management DDL on that database.
+
+We recommended that you run all role management commands in one
+database.
+
+If role replication is turned off, then the administrator must ensure that
+any roles used by DDL on one node also exist on the other nodes. Otherwise PGD apply
+stalls with an error until the role is created on the other nodes.
+
+!!! Note PGD with non-PGD-enabled databases 
```

Comment:
I wonder whether we need to consider multi-PGD-databases as well here ... maybe just "Note PGD with other databases"?

I.e. in a scenario with nodes A <-> B <-> C, where A and B form one PGD cluster and B and C form another one, role replication from A would reach only B, but not C (somewhat obviously).

I guess these kind of complications are why we de-support multi-PGD-database support.

Probably good as is with multi-database-support not supported.

Reply:
I think we need to focus the docs on the baseline common configurations and extend them where needed into what are currently edge cases as we hit them.